tests/subsys/net/lib/wifi_credentials/src/main.c

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/fff.h>
+
+#include <net/wifi_credentials.h>
+
+#include "wifi_credentials_internal.h"
+
+DEFINE_FFF_GLOBALS;
+
+FAKE_VALUE_FUNC(int, wifi_credentials_store_entry, size_t, const void *, size_t)
+FAKE_VALUE_FUNC(int, wifi_credentials_load_entry, size_t, void *, size_t)
+FAKE_VALUE_FUNC(int, wifi_credentials_delete_entry, size_t)
+FAKE_VALUE_FUNC(int, wifi_credentials_backend_init)
+
+uint8_t fake_settings_buf[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES][ENTRY_MAX_LEN];
+
+int custom_wifi_credentials_store_entry(size_t idx, const void *buf, size_t buf_len)
+{
+	TEST_ASSERT(idx < CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES);
+	memcpy(fake_settings_buf[idx], buf, MIN(ENTRY_MAX_LEN, buf_len));
+	return 0;
+}
+
+int custom_wifi_credentials_load_entry(size_t idx, void *buf, size_t buf_len)
+{
+	TEST_ASSERT(idx < CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES);
+	memcpy(buf, fake_settings_buf[idx], MIN(ENTRY_MAX_LEN, buf_len));
+	return 0;
+}
+
+void setUp(void)
+{
+	RESET_FAKE(wifi_credentials_store_entry);
+	RESET_FAKE(wifi_credentials_load_entry);
+	RESET_FAKE(wifi_credentials_delete_entry);
+	wifi_credentials_store_entry_fake.custom_fake =
+		custom_wifi_credentials_store_entry;
+	wifi_credentials_load_entry_fake.custom_fake =
+		custom_wifi_credentials_load_entry;
+}
+
+#define SSID1 "test1"
+#define PSK1 "super secret"
+#define SECURITY1 WIFI_SECURITY_TYPE_PSK
+#define BSSID1 "abcdef"
+#define FLAGS1 WIFI_CREDENTIALS_FLAG_BSSID
+
+#define SSID2 "test2"
+#define PSK2 NULL
```

Comment:
Maybe just pass NULL directly where PSK2 is used, easy to forget that PSK2 is NULL